### PR TITLE
CAM SE and cam-common-code

### LINKED ISSUE #343: Feature request: CAM SE Manhattan
### Use case  

Manhattan release for CAM-SE

### Is your feature request related to a problem?  

* cam-se not released for Manhattan
   * :bug: watch out for indices swap here: #246
* Duplicated code across cam-fv and cam-se

### Describe your preferred solution  

I believe this is the code that people have been using for cam-se using the cam-common-code:
https://github.com/jlaucar/DART/tree/CAM_SE

Brought up to date with main here:
https://github.com/NCAR/DART/tree/CAM_SE
Be aware:  index swap #246

Tidy (not changed the code): https://github.com/NCAR/DART/tree/cam-fv-se-shared f93ba7d7d674941de2d0b82a8f3024bc43a43f2f
 * model directories
    * cam-fv 
    * cam-se
    * cam-common-code
 * svn junk removal
 * path_names
 * obsolete file removal

https://github.com/NCAR/DART/compare/CAM_SE...cam-fv-se-shared

### PR DESCRIPTION
**Pull request by hkershaw-brown**
## Description:

The cam-se model_mod has been used, but had not made it to the main DART repo for release. 
The cam-se model_mod makes use of cam-common-code  module.  In this pull request, cam-fv uses the cam_common_code_mod. 

### Fixes issue

Fixes #343 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.

### Tests
Please describe any tests you ran to verify your changes.

cam-se
cam-fv   vs. main

Needs code coverage check to run all the various CAM options.

## Checklist for merging

- [x] Updated changelog entry
- [x] Documentation updated
       Documentation needs info about CESM options to generate output required for DART.

       cat user_nl_cam 
       ! Users should add all user specific namelist changes below in the form of 
       ! namelist_var = new_namelist_value 
       inithist               = 'ENDOFRUN' 
       se_write_all_corners = .true.
       

- [ ] Version tag 

## Testing Datasets

- [x] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [ ] No dataset needed

### CODE REVIEW

**Comment by nancycollins** [comment]
the original quad interp routines used an 2d array with the dimensions ordered in a way that seemed the most straightforward for the calling code.  jeff s. swapped the dimension order.  if i remember right this was based on some compiler warnings that it was going to have to make a temp array on the stack.  i believe the impact to performance would be undetectable and so the change wasn't necessary.  in any case, not all calls to the routine were switched leading to core dumps in the best case and wrong results in the worst.  this is a terrible thing to change because the compiler can't catch it. now either all new code has to use the changed order, or the order needs to be switched back.  however switching back leaves users with the very real danger of mixing and matching versions that are inconsistent.  i really don't know the right way forward here.

**Comment by hkershaw-brown** [comment]
Yeah I am in two minds about this.  Jeff A. did the CAM-SE-FV work before the Jeff S.  swap. 
The Jeff A. version has the same order as the Reanalysis branch. So stay as close as possible to main, or to the reanalysis branch (which I think has been run more than main).

Devil and the deep blue sea.  I'm leaning towards staying as close as possible to main (removing the swap back from this pull request).  I think the test is easy, "does the code blow up if ens_size > 4 ?" #246

File: models/cam-fv/model_mod.f90 [hunk 1]
```diff
@@ -1492,7 +1362,7 @@ subroutine cam_height_levels(ens_handle, ens_size, lon_index, lat_index, nlevels
    ! compute the height columns for each ensemble member - passing mbar() array in.
    do imember = 1, ens_size
       call build_heights(nlevels, surface_pressure(imember), surface_elevation(1), &
-                         tv(:, imember), height_array(:, imember), mbar=mbar(:, imember))
+                         pressure(:, imember), tv(:, imember), height_array(:, imember), mbar=mbar(:, imember))
```
Comment (hkershaw-brown) [inline models/cam-fv/model_mod.f90:1365]: Why is this different?

**Review by kdraeder — APPROVED** [review]
I haven't looked through the changes, as advised by @jlaucar and @hkershaw-brown.

**Pull request merged**